tests/study_tests/test_study.py

```diff
@@ -799,9 +799,21 @@ def _objective(trial: Trial) -> float:
 def test_optimize_with_progbar_parallel(capsys: _pytest.capture.CaptureFixture) -> None:
 
     study = create_study()
-    with pytest.warns(UserWarning, match="Progress bar only supports serial execution"):
-        study.optimize(lambda _: 1.0, n_trials=10, show_progress_bar=True, n_jobs=-1)
+    study.optimize(lambda _: 1.0, n_trials=10, show_progress_bar=True, n_jobs=2)
+    _, err = capsys.readouterr()
+
+    # Search for progress bar elements in stderr.
+    assert "10/10" in err
+    assert "100%" in err
 
+
+def test_optimize_with_progbar_parallel_timeout(capsys: _pytest.capture.CaptureFixture) -> None:
```

Comment:
I found that the first assertion (i.e., `assert "10/10" not in err` ) in this function is True even if the value of `n_jobs` is 1. How about replacing the assertions with `assert "|" not in err`? It can be seen in `test_optimize_progbar_no_constraints` and it ensures that any progress-bar-related messages are not shown.

```diff
-    assert "10/10" not in err
-    assert "100%" not in err
+    # Testing for a character that forms progress bar borders.
+    assert "|" not in err
```

Reply:
Right, fixed.

Reply:
Thanks!